### PR TITLE
Fix adjoint tests and generator

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** The Python test-suite was not fully testing the generators or adjoint pipeline for the newly added gates.

**Description of the Change:** This ensures all tests go through the adjoint pipeline.

**Benefits:** Allows incorrect gates and generator kernels to be caught early.

**Possible Drawbacks:** None

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp [hunk 1]
```diff
@@ -64,56 +80,40 @@ class StateVectorCudaManaged
...
         BaseType::CopyGpuDataToGpuIn(other);
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecCreate(
-            /* custatevecHandle_t* */ &handle));
     }
 
-    ~StateVectorCudaManaged() {
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecDestroy(
-            /* custatevecHandle_t */ handle));
-    }
+    ~StateVectorCudaManaged() {}
```
Comment (chaeyeunpark) [inline pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:116]: How about
```suggestion
    ~StateVectorCudaManaged() = default;
```
or just remove the definition?

**Review by chaeyeunpark — APPROVED** [review]
Thanks, @mlxd for the nice work! 

I just have one small suggestion.

File: pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp (hunk 1 above)
Comment (mlxd) [reply pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:116]: Yep, `default` is fine by me. Thanks @chaeyeunpark

**Review by maliasadi — APPROVED** [review]
Thanks @mlxd! Happy to approve.

**Pull request merged**